src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -427,5 +382,112 @@ private void LogScriptIfGiven(string type, string scriptPath)
+                        this.logger.LogDebug("Overriding the post-build commands with the BuildConfigurationFile section");
+                        this.logger.LogDebug(this.cliOptions.PostBuildCommand.ToString());
+                        this.writer.WriteLine("Overriding the post-build commands with the BuildConfigurationFile section");
+                        this.writer.WriteLine("\t" + this.cliOptions.PostBuildCommand.ToString());
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.logger.LogWarning("Invalid BuildConfigurationFile " + ex.ToString());
+                    this.writer.WriteLine($"{Environment.NewLine}\"{DateTime.UtcNow.ToString("yyyy-MM-dd hh:mm:ss")}\" | WARNING | Invalid BuildConfigurationFile | Exit Code: 1 | Please review your {FilePaths.AppSvcFileName} | {Constants.BuildConfigurationFileHelp}");
+                    this.writer.WriteLine($"The following is the structure of a valid {FilePaths.AppSvcFileName}:");
+                    this.writer.WriteLine("-------------------------------------------");
+                    this.writer.WriteLine("version: 1");
+                    this.writer.WriteLine("pre-build: apt-get install xyz");
+                    this.writer.WriteLine("post-build: |");
+                    this.writer.WriteLine("  python manage.py makemigrations");
+                    this.writer.WriteLine("  python manage.py migrate");
+                    this.writer.WriteLine("-------------------------------------------");
+                }
+            }
+            else
+            {
+                this.logger.LogDebug($"No {FilePaths.AppSvcFileName} found");
+            }
+        }
+
+        /// <summary>
+        /// Checks for the <see cref="FilePaths.ExtensibleConfigurationFileName"/> file in the source repository, and
+        /// if it exists, parses the YAML file for extensibility endpoints defined by the user that will be converted
+        /// to runnable steps during the build.
+        /// </summary>
+        /// <param name="context"><see cref="BuildScriptGeneratorContext"/> object containing information regarding
+        /// the user's provided source repository.</param>
+        private string ProcessExtensibleConfigurationFile(BuildScriptGeneratorContext context)
+        {
+            var filePath = Path.Combine(context.SourceRepo.RootPath, FilePaths.ExtensibleConfigurationFileName);
+            this.logger.LogDebug($"Checking for path to {FilePaths.ExtensibleConfigurationFileName}: '{filePath}'");
+            this.writer.WriteLine($"Checking for path to {FilePaths.ExtensibleConfigurationFileName}: '{filePath}'");
+
+            if (File.Exists(filePath))
+            {
+                this.logger.LogDebug("Found extensible configuration file.");
+                this.writer.WriteLine($"{Environment.NewLine}Found extensible configuration file, {filePath}");
```

Comment:
The logger will be writing to Application Insights and the writer will be writing to the console output that the user sees. In general, the writer can be more verbose in describing files, directories, commands specific to the user, etc. since it's _only_ being written to the console output and not stored anywhere, whereas with the logger, since it's being written/stored to AI, we need to be cautious of PII that could be sent there.

Reply:
Makes sense, thanks for the explanation